impl/vfs/hdfs/src/main/java/org/pentaho/big/data/impl/vfs/hdfs/nc/NamedClusterProvider.java

```diff
@@ -199,8 +187,8 @@ private void addCacheEntry( FileName rootName, FileSystem fs ) throws FileSystem
   }
 
   public void closeFileSystem( String embeddedMetastoreKey ) {
-    IMetaStore defaultMetastore = metaStoreService.getMetastore();
-    IMetaStore embeddedMetastore = metaStoreService.getExplicitMetastore( embeddedMetastoreKey );
+    IMetaStore defaultMetastore = getMetastoreLocator().getMetastore();
+    IMetaStore embeddedMetastore = getMetastoreLocator().getExplicitMetastore( embeddedMetastoreKey );
```

Comment:
this will NPE if getMetastoreLocator() doesn't find the plugin for whatever reason. Is that the expected behavior?

Reply:
Yes; if the metastore isn't found, we have a big problem and should definitely error out.